backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/impl/PurchaseOrderLineModularContractHandler.java

```diff
@@ -176,4 +190,32 @@ public void validateDocAction(
 			throw new AdempiereException(MSG_REACTIVATE_NOT_ALLOWED);
 		}
 	}
+
+	@Override
+	public void cancelLinkedContractsIfAllowed(@NonNull final I_C_OrderLine model, @NonNull final FlatrateTermId flatrateTermId)
+	{
+		final List<I_M_InOutLine> generatedInOutLines = inOutDAO.retrieveLinesForOrderLine(model);
+
+		if (!generatedInOutLines.isEmpty())
+		{
+			final Set<Integer> inoutIds = generatedInOutLines.stream()
+					.map(I_M_InOutLine::getM_InOut_ID)
+					.collect(ImmutableSet.toImmutableSet());
+
+			throw new AdempiereException(MSG_VOID_NOT_ALLOWED, inoutIds);
+		}
+
+		final I_C_Flatrate_Term contract = flatrateDAO.getById(flatrateTermId);
+
+		// dev-note: for now set fallback endDate one year from start date
+		final Timestamp endDate = Optional.ofNullable(contract.getEndDate())
+				.orElse(TimeUtil.addYears(contract.getStartDate(), 1));
+
+		final IContractChangeBL.ContractChangeParameters parameters = IContractChangeBL.ContractChangeParameters.builder()
+				.changeDate(endDate)
+				.action(ChangeTerm_ACTION_VoidSingleContract)
+				.build();
```

Comment:
I would prefer here to use action ChangeTerm_ACTION_Cancel.
Although ChangeTerm_ACTION_VoidSingleContract si canceling the current and in this case shall be enough because such a modular contract shall never be extended.

Reply:
Ok